### PR TITLE
Clarify error message of torch.nn.functional.embedding_bag

### LINKED ISSUE #99221: the error message of torch.nn.functional.embedding_bag is not clear
### 🐛 Describe the bug

```
import torch
arg_1 = torch.as_tensor([2])
arg_2 = torch.rand([2, 3], dtype=torch.float32)
arg_3 = torch.as_tensor([0])
res = torch.nn.functional.embedding_bag(input=arg_1,weight=arg_2,offsets=arg_3)
print(res)
```
Above code will throw an exception: `RuntimeError: Index 0 is out of bounds: 2, range 0 to 2`. This error message is not clear, because the range of `Index 0` is greater than or equal to 0 and less than 2. The message `range 0 to 2` is a little confused and people will think that `Index 0` can take the value of 2.

### Versions

pytorch version: 2.0.0+cu118

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @malfet

### PR DESCRIPTION
**Pull request by tfsingh**
Fixes #99221  , clarifying the error message to highlight the index from inputs which is responsible for the out-of-bounds error,  while maintaining the reference to the relevant index of offsets as a secondary consideration. 

Also takes care of some spelling/grammatical issues with another message (primarily "yout" changed to "your").

Would be happy to iterate on this if there are any issues.

cc @jbschlosser

### CODE REVIEW

File: aten/src/ATen/native/EmbeddingBag.cpp [hunk 1]
```diff
@@ -177,18 +177,19 @@ void fbgemm_spmdm_report_error_(
       index_t idx = indices[i];
       TORCH_CHECK(
           0 <= idx && idx < N,
-          "Index ",
-          i,
-          " is out of bounds: ",
+          "Index value ",
```
Comment (albanD) [inline aten/src/ATen/native/EmbeddingBag.cpp:180]: Another possible phrasing would be "The {idx}-th index value is {i} which is not in the valid range[0, {N}[" which I feel would be clearer no?

Comment (tfsingh) [reply aten/src/ATen/native/EmbeddingBag.cpp]: Yes, certainly -- I committed a new version which should capture the goal of your comments (note -- I believe you might have idx and i switched in your suggestion).

Thank you so much for your review, please let me know if this requires further iteration.

File: aten/src/ATen/native/EmbeddingBag.cpp [hunk 2]
```diff
@@ -179,16 +179,17 @@ void fbgemm_spmdm_report_error_(
...
           "Index ",
           i,
-          " is out of bounds: ",
+          " of input takes value ",
           idx,
-          ", range 0 to ",
-          N);
+          " which is not in the valid range [0, ",
+          N,
+          "[");
```
Comment (mikaylagawarecki) [inline aten/src/ATen/native/EmbeddingBag.cpp:186]: nit: exclusive bracket for N

```suggestion
          " which is not in the valid range [0, ",
          N,
          ")");
```

Comment (tfsingh) [reply aten/src/ATen/native/EmbeddingBag.cpp]: Of course, just fixed.

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**